Products/PasswordStrength/backward.py

```diff
@@ -0,0 +1,64 @@
+from Products.PluggableAuthService.interfaces.plugins import IValidationPlugin
+from Products.CMFCore.utils import getToolByName
+from Products.PasswordStrength import _
+from plone.app.users.browser.register import BaseRegistrationForm
+from plone.app.users.browser.personalpreferences import PasswordAccountPanel
+
+
+# monkey patch to be used for Plone < 4.3
+def testPasswordValidity(self, password, confirm=None):
+
+    """ Verify that the password satisfies the portal's requirements.
+
+    o If the password is valid, return None.
+    o If not, return a string explaining why.
+    """
+    if confirm is not None and confirm != password:
+        return _('Your password and confirmation did not match. Please try again.')
+
+    if not password:
+        err = [_('You must enter a password')]
+    else:
+        err = []
+
+    # We escape the test if it looks like a generated password
+    if password and password.startswith('G-') and len(password) == len(self.origGeneratePassword()) + 2:
+        return None
+
+    # Use PAS to test validity
+    pas_instance = self.acl_users
+    plugins = pas_instance._getOb('plugins')
+    validators = plugins.listPlugins(IValidationPlugin)
+    for validator_id, validator in validators:
+        user = None
+        set_id = ''
+        set_info = {'password': password}
+        errors = validator.validateUserInfo(user, set_id, set_info)
```

Comment:
The following format does a better job at explaining what's going on with this call because it explicitly indicates how the the data is being passed to the method.:

```python
errors = validators.validateUserInfo(user=None, set_id='', set_info={'password': None})
```

I'd ditch the intermediate variables since they are just hurting you from writing this call in a way that documents what data is going where.